README.md

```diff
@@ -14,52 +14,114 @@ When you are done:
 
 #### Loops & Lists
 - [Max Value](https://repl.it/@Admin7/maxvalue)
-```python
-# copy & paste your solution here
+```
+def return_max(l):
+  
+  if len(l) == 0:
+    return None 
+  
+  max_number = l[0]
+  for i in range (1, len(l)):
+    if l[i] > max_number: 
+      max_number = l[i]
+      
+  return max_number 
 ```
 
 - [Min Value](https://repl.it/@Admin7/minvalue)
 ```python
-# copy & paste your solution here
+def return_min(l):
+  if len(l) == 0:
+    return None
+  min_number = l[0] 
+  for i in range (1, len(l)):
+    if l[i] < min_number: 
+      min_number = l[i]
+  return min_number 
 ```
 
 - [Extend](https://repl.it/@Admin7/extendlist)
 ```python
-# copy & paste your solution here
+def return_extended(l, iter):
+  l.extend(iter)
+  return l
 ```
 
 - [Extend](https://repl.it/@Admin7/creatematrix)
 ```python
-# copy & paste your solution here
+def create_matrix(h, w):
+  matrix=[]
+  for i in range(h):
+    row=[]
+    for j in range(w):
+      if w == 1:
+        break 
+      value='#'
+      row.append(value)
+    matrix.append(row)
+  return matrix
 ```
 
 
 #### Dictionaries
 - [Receipt Tally](https://repl.it/@Admin7/receipttally)
 ```python
-# copy & paste your solution here
+def get_total(trans, category):
+  total = 0
+  for lineitem in trans: 
+    current_category = lineitem.get('category')
+    current_subtotal = lineitem.get('subtotal')
+    if current_category == category:
+      total += current_subtotal
+  return total
 ```
 
 - [Translation](https://repl.it/@Admin7/translations)
 ```python
-# copy & paste your solution here
+def translation(word, lang): 
+  trans_for_word = translations.get(word)
+  if (trans_for_word == None):
+    return 'Word does not exist in translations'
+  lang_translation = trans_for_word.get(lang) 
+  if (lang_translation == None):
+    return 'Language does not exist in translation'
+  return lang_translation
 ```
 
 - [Check RSVP](https://repl.it/@Admin7/checkrsvp)
 ```python
-# copy & paste your solution here
+def check_rsvp(event):
+  attendees = 0
+  for x in event.values():
+    if x == True:
+      attendees += 1
+      
+  return attendees
 ```
 
 
 #### Functions
 - [Aint No Callback Girl](https://repl.it/@Admin7/aintnocallbackgirl)
 ```python
-# copy & paste your solution here
+def callbacker(l,callback):
+  for i in range(0, len(l)):
+    if callback==double:
+      l[i]=double(l[i])
+    elif callback==add_one:
+      l[i]=add_one(l[i])
```

Comment:
Let's think about what happens when we define a function:
```python
def some_name(some_param):
  return 'some_result'

# first, we define a function name & we define some parameter names
# these both take up name spaces, just like variables do
# then we probably return something
```

I could then take that function and assign it to another variable name, but it's still that function:

```python
another_name = some_name # still the same function!!!
```
So why am I pointing this out? Because `callback` is a parameter that is assigned to whatever function we pass in. If we pass in `double`, `callback` is equal to `double`. If we pass in `add_one`, `callback` is equal to `add_one`. This means, running `callback(num)` is like calling `double(num)` or `add_one(num)`.

With that in mind, how would you rewrite this?